### ISSUE TITLE
Add a permission setting to optionally allow popups from websites

### ISSUE THREAD

**Issue by josemiguelo** [Jelmerro/Vieb#67]
Hi!
First of all, thanks for this piece of software. Just loving it so far! It feels like a web browser tailor-made for vim users. Simple, but with all the things I like from a modal context. 

Just wondering, how exactly can I deal with popup windows? My scenario is clicking a button inside a email for resetting a password. It seems like it's trying to open something but I just keep getting this message:

<img width="419" alt="Screen Shot 2020-07-29 at 15 11 42" src="https://user-images.githubusercontent.com/15330034/88848300-e2a17180-d1ad-11ea-90c4-444c8de9222d.png">

After that, a blank tab remains opened, but the requested page doesn't open at all. Not sure how to address this or where to start.

Any guidance on this? Thanks!

**Comment by Jelmerro** [comment]
I don't think that I use any site that uses popups to confirm or show any relevant information. As such, I have personally always considered the lack of popup support more a feature than a bug. I'm curious as to which site this might be and what your view on popups are, given that you clearly have more use for them than I have.

On the other hand, you could also ask the website to stop using popups to confirm a password reset. Other browsers such as Firefox have popups [blocked by default](https://support.mozilla.org/en-US/kb/pop-blocker-settings-exceptions-troubleshooting), which to me is a good thing, although it would also be great if Vieb offered the same level of customization for the popup blocker as Firefox. The difference here being that Vieb isn't explicitly blocking them, but it's more of a [default for Electron webviews](https://www.electronjs.org/docs/api/webview-tag#allowpopups). I'll just check if this webview flag can be changed for existing webviews, if so, we could think of adding settings related to popups.

**Comment by Jelmerro** [comment]
I hate to bring some bad news regarding this ticket, but I can't exactly reproduce the popup behavior with some of the sites that do apparently use popups (in Firefox). My experience has actually been that those will open in a new tab without issue. In addition to that, it appears that webviews do not allow this "allowpopups" flag to be changed after the webview was created. As a result, I don't think it's possible to make this a setting without requiring a restart, or at the very least require tabs to be re-opened. Having said that, I don't think I'll be making in effort to work on this myself. If you understand Javascript you can make your own Vieb build which permanently allows popups by going to [this line](https://github.com/Jelmerro/Vieb/blob/4daf9a16bd6e6672ce77bb136a3b3639df896e5e/app/js/tabs.js#L234) and adding this: `webview.setAttribute("allowpopups", "")`. Keep in mind that this will permanently allow popups for every single page you visit, which might not be what you want, which is exactly why I don't see this making it's way into Vieb anytime soon. Though if anyone knows a better way to integrate the popups into Vieb's permission system, go ahead.

**Comment by josemiguelo** [comment]
I'll do some research on this and try to understand it better. Thanks @Jelmerro

**Comment by Jelmerro** [comment]
I ran into this ticket again, and after some experimentation I noticed that there is no way to allow popups while also running Vieb's code in the popup window. It just opens the page itself in a new window, without any options or configuration to change this. This means that none of Vieb's features will be able to run in a popup window.

It makes more and more sense to just keep redirecting these popups to new tabs. I keep seeing this as the recommended option everywhere when they mention anything about popups, including the [window.open MDN page](https://developer.mozilla.org/en-us/docs/Web/API/Window/open#usability_issues) and the previously mentioned sources.

It's for the reasons above that Vieb will not be receiving a popup permission or allow any kind of popups in the near future. I'll close this ticket as wont fix. For those that for some reason still want to enable popups in Vieb, you'll have to download the source and make your own Vieb build which permanently allows popups by adding: `webview.setAttribute("allowpopups", "")` below [this line](https://github.com/Jelmerro/Vieb/blob/504431e202096824f6ab78242e39ab12adcd1002/app/js/tabs.js#L329).
